IDR/production/2.visualize_screen_metadata.py

```diff
@@ -0,0 +1,281 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# ## Visualize Screen Metadata
+#
+# Each screen is a unique experiment. Visualize the diversity of experimental details in IDR screens.
+#
+# **Note:** This notebook is WIP
+
+# In[1]:
+
+
+import pathlib
+import pandas as pd
+import plotnine as gg
+import os
+
+
+def describe_df(df, column_name, data_dir_str, gene_phenotype=True):
+    """Describes the number of entries and unique entries of a Pandas dataframe.
+
+    Parameters
+    ----------
+    df: Pandas dataframe
+
+    column_name: str
+        Name of the dataframe column to be described.
+
+    gene_phenotype: bool
+        True if the user wants to ensure each image has a gene_identifier and corresponding phenotype identifier
+
+    data_dir_str: str
+        Directory to store output .tsv files
+
+    Returns
+    -------
+    Table of the counts and unique entries of the dataframe.
+    """
+
+    print(f'\nVariable: {column_name}')
+    if gene_phenotype:
+        df_naOmit = df.dropna(subset=['gene_identifier', 'phenotype_identifier'])
+    else:
+        df_naOmit = df.dropna(subset=[column_name])
+
+    print(df_naOmit[column_name].describe())
+    print('___________________________')
+
+    sorted_dfs_dir = data_dir_str + "/sorted_dfs"
+
+    if not os.path.exists(sorted_dfs_dir):
+        os.mkdir(sorted_dfs_dir)
+
+    out_file = pathlib.Path(sorted_dfs_dir, f"{column_name}_df_naOmit.tsv")
+    df_naOmit.to_csv(out_file, index=False, sep="\t")
+
+    return df_naOmit
+
+# In[2]:
+
+
+data_dir = pathlib.Path("publicly-available-microscopy-data/IDR/data")
+data_dir_str = "publicly-available-microscopy-data/IDR/data"
+
+
+# ## Load data
+
+# In[3]:
+
+
+# Load IDR ids
+id_file = pathlib.Path(data_dir, "idr_ids.tsv")
+
+id_df = pd.read_csv(id_file, sep="\t")
+
+print(id_df.shape)
+id_df.head(2)
+
+
+
+# In[4]:
+
+
+# Load screen details
+screen_file = pathlib.Path(data_dir, "screen_details.tsv")
+screen_df = pd.read_csv(screen_file, sep="\t")
+
+print(screen_df.shape)
+screen_df.head(2)
+
+
+# In[5]:
+
+
+# Load all plates
+plate_file = pathlib.Path(data_dir, "plate_details_per_screen.tsv")
+
+plate_df = pd.read_csv(plate_file, sep="\t").merge(screen_df, on="screen_id", how="left")
+plate_df.n_wells = plate_df.n_wells.astype(str)
+
+print(plate_df.shape)
+plate_df.head(2)
+
+
+# Descriptive Statistics
+column_list = ['gene_identifier', 'phenotype_identifier', 'channels', 'imaging_method']
+for i in column_list:
+    describe_df(plate_df, i, data_dir_str=data_dir_str, gene_phenotype=True)
+
+plate_df_naOmit = describe_df(plate_df, 'gene_identifier', data_dir_str=data_dir_str, gene_phenotype=True)
+
+# Number of wells per experiment
+print(f"The total number of wells is: {plate_df_naOmit.n_wells.astype(int).sum():,d}")
+
+mean_num_plates = plate_df_naOmit.groupby("idr_name")["screen_id"].count().mean()
+print(f"The mean number of plates is: {mean_num_plates}")
+
+# gg.options.figure_size = (6.4, 6.4)
```

Comment:
Perhaps this script isn't ready for production?

Also, any reason the scripts are named the same as the notebooks?

Maybe we should chat more about how you're thinking about "production"

Reply:
The `2.visualize_screen_metadata.py` file should be removed from this current pull request.

Regarding your file naming comment, it's my understanding that the notebook files are used for prototyping the production files. Why would we name them something different if they are directly related to each other and perform essentially the same functionalities? The only difference is that one (notebook file) is split into chunks. My understanding of the difference between prototype and production files is probably still somewhat skewed.